comp/core/workloadmeta/collectors/internal/containerd/image_sbom_trivy.go

```diff
@@ -45,7 +45,14 @@ func (c *collector) startSBOMCollection(ctx context.Context) error {
 		workloadmeta.NormalPriority,
 		workloadmeta.NewFilter(&filterParams),
 	)
-	resultChan := make(chan sbom.ScanResult, 2000)
+	scanner := collectors.GetContainerdScanner()
+	if scanner == nil {
+		return fmt.Errorf("error retrieving global docker scanner")
```

Comment:
you are mentioning docker but it is a containerd scanner

```suggestion
		return fmt.Errorf("error retrieving global containerd scanner")
```

Reply:
is it possible that `scanner` is `nil`? 

can we split this function in two:
* `initSBOMCollection()` responsible of setting the instances of `c.sbomScanner` and `scanner`, setting the result channel  and call this function in `NewCollector()`
* `startSBOMCollection()` responsible to start the collection (register to workload) start the go routine.